external/fv3fit/tests/training/test_main.py

```diff
@@ -216,9 +289,83 @@ def get_config(tmpdir, mock_dataset, additional_variables, derived_output_variab
 
     args = MainArgs(
         data_path=data_path,
-        train_config=training_filename,
-        train_data_config=train_data_filename,
-        val_data_config=validation_data_filename,
+        training_config=training_filename,
+        training_data_config=train_data_filename,
+        validation_data_config=validation_data_filename,
         output_path=output_path,
     )
-    return args, input_variables, output_variables, hyperparameters, output_path
+    return TestConfig(
+        args, input_variables, output_variables, output_path, mock_dataset
+    )
+
+
+def cli_main(args: MainArgs):
+    if args.validation_data_config is None:
```

Comment:
I don't think the case that `args.validation_data_config` is `None` is actually tested by the parameterization. Would be good to check that.

Reply:
Fixed. This was untested in the main routines as well.